`Run Python Tests` workflow fails on setup pyenv

`Run Python Tests` workflow exits with an error on the step for pyenv installation. The error is the duplicate installation of the same Python version. The following is a log output on the `setup pyenv` from a workflow run.

```
Run gabrielfalcao/pyenv-action@v18
downloading https://github.com/pyenv/pyenv/archive/v2.4.20.zip
/usr/bin/unzip -o -q /opt/hostedtoolcache/pyenv-archive.zip/2.4.20/x[6](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:7)4/pyenv-archive.zip
Extracted /opt/hostedtoolcache/pyenv-archive.zip/2.4.20/x64/pyenv-archive.zip to /home/runner/work/_temp/[7](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:8)7350c64-cfd1-4f9e-ade6-71e252c582c7.
archive_path /opt/hostedtoolcache/pyenv-archive.zip/2.4.20/x64/pyenv-archive.zip
pyenv_root /opt/hostedtoolcache/pyenv_root/2.4.20/x64
export PYENV_ROOT="/opt/hostedtoolcache/pyenv_root/2.4.20/x64"
Patched PATH with "/opt/hostedtoolcache/pyenv_root/2.4.20/x64/bin"
/opt/hostedtoolcache/pyenv_root/2.4.20/x64/bin/pyenv install 3.12.[8](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:9)
/opt/hostedtoolcache/pyenv_root/2.4.20/x64/bin/pyenv install 3.13.1
/opt/hostedtoolcache/pyenv_root/2.4.20/x64/bin/pyenv install 3.13.1
Downloading Python-3.12.8.tar.xz...
Downloading Python-3.13.1.tar.xz...
-> https://www.python.org/ftp/python/3.13.1/Python-3.13.1.tar.xz
-> https://www.python.org/ftp/python/3.12.8/Python-3.12.8.tar.xz
Downloading Python-3.13.1.tar.xz...
-> https://www.python.org/ftp/python/3.13.1/Python-3.13.1.tar.xz
Installing Python-3.12.8...
Installing Python-3.13.1...
Installing Python-3.13.1...
BUILD FAILED (Ubuntu 24.04 using python-build 20180424)
Inspect or clean up the working tree at /tmp/python-build.2025050812213[9](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:10).2168
Results logged to /tmp/python-build.20250508122139.2168.log
Last 10 log lines:
	echo "Creating directory /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/lib/pkgconfig"; \
	/usr/bin/install -c -d -m 755 /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/lib/pkgconfig; \
fi
if test -f /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin/python3 -o -h /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin/python3; \
then rm -f /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin/python3; \
else true; \
fi
(cd /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin; ln -s python3.13 python3)
ln: failed to create symbolic link 'python3': File exists
make: *** [Makefile:2409: bininstall] Error 1
BUILD FAILED (Ubuntu 24.04 using python-build 20180424)
Inspect or clean up the working tree at /tmp/python-build.20250508122139.2170
Results logged to /tmp/python-build.20250508122139.2170.log
Last [10](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:11) log lines:
	(cd /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin; ln -s python3.13-32 python3-32) \
fi
if test "x" != "x" ; then \
	rm -f /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin/python3-intel64; \
	(cd /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/bin; ln -s python3.13-intel64 python3-intel64) \
fi
rm -f /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/share/man/man1/python3.1
(cd /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.13.1/share/man/man1; ln -s python3.13.1 python3.1)
ln: failed to create symbolic link 'python3.1': No such file or directory
make: *** [Makefile:2461: maninstall] Error 1
Failed to install python 3.13.1
Error: The process '/opt/hostedtoolcache/pyenv_root/2.4.20/x64/bin/pyenv' failed with exit code 1
Failed to install python 3.13.1
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.[12](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:13).8/lib/python3.12/bz2.py", line 17, in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
ModuleNotFoundError: No module named '_bz2'
WARNING: The Python bz2 extension was not compiled. Missing the bzip2 lib?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'readline'
WARNING: The Python readline extension was not compiled. Missing the GNU readline lib?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/hostedtoolcache/pyenv_root/2.4.[20](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:21)/x64/versions/3.12.8/lib/python3.12/lzma.py", line 27, in <module>
    from _lzma import *
ModuleNotFoundError: No module named '_lzma'
WARNING: The Python lzma extension was not compiled. Missing the lzma lib?
Installed Python-3.12.8 to /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.12.8
Sucessfully installed python 3.12.8
/usr/bin/bash /opt/hostedtoolcache/pyenv-2.4.20-command-3.12.8/3.12.8/x64/pyenv_post_install.sh
+ pip install -U pip
Requirement already satisfied: pip in /opt/hostedtoolcache/pyenv_root/2.4.20/x64/versions/3.12.8/lib/python3.12/site-packages ([24](https://github.com/Financial-Fraud-Detection-System/financial-security-system/actions/runs/14906330416/job/41869471415?pr=1#step:3:25).3.1)
Collecting pip
  Downloading pip-25.1.1-py3-none-any.whl.metadata (3.6 kB)
Downloading pip-25.1.1-py3-none-any.whl (1.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 69.1 MB/s eta 0:00:00
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 24.3.1
    Uninstalling pip-24.3.1:
      Successfully uninstalled pip-24.3.1
Successfully installed pip-25.1.1
```

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -18,6 +18,15 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Install required build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            make build-essential libssl-dev zlib1g-dev \
+            libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+            libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev \
+            liblzma-dev
+
       - name: setup pyenv
         uses: gabrielfalcao/pyenv-action@v18
         with:

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -34,15 +34,6 @@ jobs:
           versions: 3.13.1, 3.12.8
           command: pip install -U pip
 
-      - name: Cache virtual environments
-        uses: actions/cache@v3
-        with:
-          path: |
-            **/.venv
-          key: ${{ runner.os }}-venv-${{ hashFiles('**/requirements.txt', '**/.pythonversion') }}
-          restore-keys: |
-            ${{ runner.os }}-venv-
-
       - name: Install dependencies
         run: |
           # Make the script executable

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -31,7 +31,7 @@ jobs:
         uses: gabrielfalcao/pyenv-action@v18
         with:
           default: 3.13.1
-          versions: 3.13.1, 3.12.8
+          versions: 3.12.8
           command: pip install -U pip
 
       - name: Install dependencies

